File: frontend/src/domain/dtos/FlashCardLearningStateDto.ts
import { LearningTaskType } from "../LearningTaskType";

// Keep in sync with backend/VocabuAI.Api/Dtos/FlashCardLearningStateDto.cs
export type FlashCardLearningStateDto = {
  id: number;
  flashCardId: number;
  box: number;
  progressPointsInCurrentBox: number;
  correctCountsByQuestionTypeInCurrentBox: Record<LearningTaskType, number>;
  correctCountTotal: number;
  wrongCountTotal: number;
  correctStreak: number;
  lastAnsweredAt: string | null;
};

File: backend/VocabuAI.Api/Dtos/FlashCardLearningStateDto.cs
using VocabuAI.Domain.Learning;

namespace VocabuAI.Api.Dtos;

// Keep in sync with frontend/src/domain/dtos/FlashCardLearningStateDto.ts
public sealed record FlashCardLearningStateDto(
    int Id,
    int FlashCardId,
    int Box,
    int ProgressPointsInCurrentBox,
    Dictionary<LearningTaskType, int> CorrectCountsByQuestionTypeInCurrentBox,
    int CorrectCountTotal,
    int WrongCountTotal,
    int CorrectStreak,
    DateTimeOffset? LastAnsweredAt);
